### PR TITLE
Make `docs` have its own `pyproject.toml` file to simplify dependencies specification.

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
Also update the Docker file accordingly

### CODE REVIEW

File: docs/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[tool.poetry]
+name = "parsec-docs"
+version = "0.0.0"
+description = "The parsec-cloud documentations"
```
Comment (mmmarcos) [inline docs/pyproject.toml:4]:
```suggestion
description = "Parsec documentation"
```

File: docs/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+authors = ["Scille SAS <contact@scille.fr>"]
+license = "BUSL-1.1"
+readme = "../README.md"
+packages = []
+
+[tool.poetry.dependencies]
+python = "^3.9"
```
Comment (mmmarcos) [inline docs/pyproject.toml:11]: Should this file also be added to the `/misc/version_updater.py` script? (it contains a Python version)

File: server/pyproject.toml
Comment (mmmarcos) [inline server/pyproject.toml:1]: Since this is moved to `/docs/pyproject.toml`, does dependabot configuration needs to be updated?

File: make.py [hunk 3]
```diff
@@ -160,7 +160,7 @@ def run(self, cwd: Path, extra_cmd_args: Iterable[str]) -> None:
         # (note only the first `maturin develop` is impacted, as all maturin
         # develop uses the same default virtualenv)
         Cmd(
-            cmd="poetry install --with=docs,testbed-server",
+            cmd="poetry install --with=testbed-server",
```
Comment (mmmarcos) [inline make.py:163]: No need to install docs in `make.py`? btw, maybe we should create an issue to add a `$ ./make.py docs` option.

Comment (FirelightFlagboy) [reply make.py:163]: Both `pyproject.toml` with poetry will not share the same `virtualenv`.

IMO very few persons need the docs dependencies so maybe a `make.py docs` that do a `poetry -C docs install` could do the trick

File: server/pyproject.toml
Comment (FirelightFlagboy) [reply server/pyproject.toml:1]: Done, I've choose to make dependabot to update those dependencies in a monthly basis since they're not as important

**Review by mmmarcos — APPROVED** [review]
LGTM

File: make.py (hunk 3 above)
Comment (FirelightFlagboy) [reply make.py:163]: I've created the issue #5356 to track that

**Pull request merged**